spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/data/TestHelpers.java

```diff
@@ -113,33 +113,6 @@ public static void assertEqualsSafe(Types.StructType struct, Record rec, Row row
     }
   }
 
-  public static void assertEqualsBatch(
```

Comment:
No longer needed, it's public but it's in tests in Spark so I think we're OK to directly remove this? Don't think we're breaking any compatibilities guarantees the project provides by removing it.

Reply:
this should be fine to be removed, since it's not used anywhere and we don't provide API guarantees for tests